tests/ci_tests/code_formatter/crewai/test_crewai_agents.py

```diff
@@ -0,0 +1,310 @@
+        assert isinstance(agent.tools[0], PostSuggestionToolLangAI)
+
+    def test_reviewer_does_not_delegate(self):
+        """Test that the reviewer has delegation disabled"""
+        mock_langfuse_client = mock.MagicMock()
+
+        agent_map = agents.create_pr_reviewer_agent(mock_langfuse_client)
+        agent = agent_map["pull_request_reviewer"]
+
+        assert agent.allow_delegation is False
+
+    def test_reviewer_is_verbose(self):
+        """Test that the reviewer has verbose mode enabled"""
+        mock_langfuse_client = mock.MagicMock()
+
+        agent_map = agents.create_pr_reviewer_agent(mock_langfuse_client)
+        agent = agent_map["pull_request_reviewer"]
+
+        assert agent.verbose is True
+
+    def test_reviewer_does_not_use_langfuse_prompts(self):
+        """Test that the reviewer doesn't call Langfuse (uses hardcoded prompts)"""
+        mock_langfuse_client = mock.MagicMock()
+
+        agents.create_pr_reviewer_agent(mock_langfuse_client)
+
+        # The reviewer agent fetches a template twice to build its goal.
+        mock_langfuse_client.get_prompt.assert_called_with("reformatted_code_file_template")
+        assert mock_langfuse_client.get_prompt.call_count == 2
+
+
+class TestAgentIntegration:
+    """Integration tests for agent creation"""
+
+    def test_both_agents_can_be_created_together(self):
+        """Test that both agent creation functions work together"""
+        mock_langfuse_client = mock.MagicMock()
+
+        mock_goal_prompt = mock.MagicMock()
+        mock_goal_prompt.prompt = "goal"
+        mock_backstory_prompt = mock.MagicMock()
+        mock_backstory_prompt.prompt = "backstory"
+
+        mock_langfuse_client.get_prompt.side_effect = [
+            mock_goal_prompt,
+            mock_backstory_prompt,
+            "template",
+            "template",
+        ]
+
+        formatter_agents = agents.create_code_formatter_agents(mock_langfuse_client)
+        reviewer_agents = agents.create_pr_reviewer_agent(mock_langfuse_client)
+
+        assert "senior_engineer" in formatter_agents
+        assert "pull_request_reviewer" in reviewer_agents
+
+        # They should be different agents
+        assert formatter_agents["senior_engineer"] != reviewer_agents["pull_request_reviewer"]
+
+    def test_agents_can_be_merged(self):
+        """Test that agent dicts can be merged (as done in main.py)"""
+        mock_langfuse_client = mock.MagicMock()
+
+        mock_goal_prompt = mock.MagicMock()
+        mock_goal_prompt.prompt = "goal"
+        mock_backstory_prompt = mock.MagicMock()
+        mock_backstory_prompt.prompt = "backstory"
+
+        mock_langfuse_client.get_prompt.side_effect = [
+            mock_goal_prompt,
+            mock_backstory_prompt,
+            "template",
+            "template",
+        ]
+
+        agent_map = agents.create_code_formatter_agents(mock_langfuse_client)
+        agent_map.update(agents.create_pr_reviewer_agent(mock_langfuse_client))
```

Comment:
Added a docstring to the method, describing its purpose.
```suggestion
    def test_agents_can_be_merged(self):
```